mozilla/nucleus
# Upgrade to Django 4.2.x and Python 3.12.x

This changeset moves Nucleus up to Python and Django versions that are not too close to EOL.

The move to Python 3.12.1 also meant moving away from gunicorn+meinheld, as meinheld [won't compile above Python 3.9](https://github.com/mozilla/bedrock/issues/13247)

While in the neighbourhood, I've also switched to `ruff` from `isort` + `flake8`



## To test:

* check out branch
* `make clean build test`
* `docker-compose up web`
* `make run-shell` then `./manage.py createsuperuser` to be able to log in
* Go to http://localhost:8000 and log in to the nucleus admin
* add a release or two, with some notes to confirm all works ok 

## Other considerations:

* We can also test this on nucleus dev once it's hit `master`, but there's no test env. Suggestions/recollections for how this has been tested in the past are welcome.
